src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedUberStore.scala

```diff
@@ -85,9 +85,10 @@ object SegmentedUberStore {
 
     repair(base)
 
+    def buildSegment(base: JFile, name: String) = Segment(siriusConfig, base, name)
     val segmentedCompactor = SegmentedCompactor(siriusConfig)
 
-    new SegmentedUberStore(new JFile(base), MAX_EVENTS_PER_SEGMENT, segmentedCompactor)
+    new SegmentedUberStore(new JFile(base), MAX_EVENTS_PER_SEGMENT, segmentedCompactor, buildSegment)
```

Comment:
Ok, so here's a thought. We've been trying to get the config weaved down nicely to the bottom layers so we can build the right read/write handles, according to that config. Instead of going to all that trouble, how about we just build an `UberFileDesc` (maybe with a rename??) _once_ here... or even up in SiriusFactory... based on the config. The whole job of that thing is to build the right read/write handles, and to do it on demand. Building one of those that is used by all the different segments makes sense I think.

We can pass that down wherever it needs to go. Segments can just accept that as a parameter, since it's all they're doing anyway, and we can get rid of some of these config-hiding abstractions like passing in `buildSegment` it also limits the scope of the config creep.

Reply:
So instead of each instance of `UberFileDesc` (name pending) existing per data file there'd be one shared instance that can open a handle to any file?

Reply:
Yeah, something along those lines. All it does is take file paths and build the right kinds of input streams / output file handles, based on configured behavior. None of the lower-level pieces (Segment, etc) care about _how_ it does its job. They just need something that can get them a `.foreach` from a file. Passing in something from above that can do that translation (sort of analogous to the way we do the event codec) makes sense imo.

Reply:
ok, I've pushed a commit which includes some of the refactoring.  I went with the name `UberStoreFileHandleFactory` and made it a trait.  I've not yet touched the unit tests and it needs scaladocs but I wanted to get input on the refactoring first.

Reply:
I've refactored a bit more around.  I changed some of the names around to be (what I think is) more consistent.  So the new trait is `UberDataFileHandleFactory` and it can create `UberDataFileReadHandle` and `UberDataFileWriteHandle` instances.  I still pass the `buildSegment` function around but I could easily refactor that away, although I like the idea that it's kept abstracted away and only `SegmentedUberStore` knows how its implemented.

I did update the existing unit tests this time and they all pass but this would require some additional tests, plus scaladocs, but again I'd like some feedback as to the direction.

Reply:
I like it & I have no other comments on how it is right now beyond what you lined out already 👍